[flatbuffers] downstream deps break (e.g. mnn) if Abseil is installed

**Host Environment**

- OS: Linux
- Compiler: GCC 12

**To Reproduce**

Steps to reproduce the behavior:

```
./vcpkg install --binarysource=clear abseil mnn
```

note that installing just `mnn` works:

```
./vcpkg remove --recurse abseil mnn
./vcpkg install --binarysource=clear  mnn
```


**Failure logs**

```
error: building mnn:x64-linux failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mnn
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[mnn]+Build+error
Include '[mnn] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-02-16-12e657924d99511514c0287ca5ce46882d3657c7
    vcpkg-scripts version: 64fabc940 2023-03-02 (17 hours ago)
```

<summary>
<details>Compiler error</details>
```
/usr/bin/c++ -DDEBUG -DMNN_DEBUG -DMNN_DEBUG_MEMORY -DMNN_DEBUG_TENSOR_SIZE -DMNN_SUPPORT_TFLITE_QUAN -DMNN_USE_SSE -DMNN_USE_THREAD_POOL -DMNN_VERSION=\"1.1.0.0-d6795ad\" -DMNN_VERSION_MAJOR=1 -DMNN_VERSION_MINOR=1 -DMNN_VERSION_PATCH=0 -I/usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/include -I/usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/source -I/usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/express -I/usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/tools -I/usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/schema/current -I/usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/3rd_party/half -I/usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/share/rapidjson/../../include -isystem /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include -fPIC -std=c++11 -D__STRICT_ANSI__ -g -fstrict-aliasing -ffunction-sections -fdata-sections -ffast-math -fno-rtti -fno-exceptions  -g   -fPIC -std=gnu++11 -MD -MT CMakeFiles/MNNMath.dir/source/math/Matrix.cpp.o -MF CMakeFiles/MNNMath.dir/source/math/Matrix.cpp.o.d -o CMakeFiles/MNNMath.dir/source/math/Matrix.cpp.o -c /usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/source/math/Matrix.cpp
In file included from /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/base/config.h:86,
                 from /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/base/attributes.h:37,
                 from /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/strings/string_view.h:39,
                 from /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/flatbuffers/base.h:237,
                 from /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/flatbuffers/array.h:22,
                 from /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/flatbuffers/flatbuffers.h:24,
                 from /usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/schema/current/Tensor_generated.h:7,
                 from /usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/source/core/TensorUtils.hpp:13,
                 from /usr/local/google/home/coryan/vcpkg-develop/buildtrees/mnn/src/1.1.0-9f04ab166e.clean/source/math/Matrix.cpp:12:
/usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/base/policy_checks.h:79:2: error: #error "C++ versions less than C++14 are not supported."
   79 | #error "C++ versions less than C++14 are not supported."
      |  ^~~~~
/usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/strings/string_view.h:52:26: error: ‘string_view’ in namespace ‘std’ does not name a type
   52 | using string_view = std::string_view;
      |                          ^~~~~~~~~~~
/usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/strings/string_view.h:52:21: note: ‘std::string_view’ is only available from C++17 onwards
   52 | using string_view = std::string_view;
      |                     ^~~
/usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/strings/string_view.h:686:8: error: ‘string_view’ does not name a type
  686 | inline string_view ClippedSubstr(string_view s, size_t pos,
      |        ^~~~~~~~~~~
/usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/include/absl/strings/string_view.h:697:11: error: ‘string_view’ does not name a type
  697 | constexpr string_view NullSafeStringView(const char* p) {
```
</summary>

**Additional context**

This is actually a problem in flatbuffers (https://github.com/google/flatbuffers/pull/7824), it magically uses Abseil headers without declaring a dependency on it.  This was actually discovered by @dg0yt in #29792.  I am filing this bug to explain the problem in more detail and justify the need for #29974.

--- a/ports/flatbuffers/fix-abseil-cxx14.patch
+++ b/ports/flatbuffers/fix-abseil-cxx14.patch
@@ -1,16 +1,3 @@
-From a56f9ec50e908362e20254fcef28e62a2f148d91 Mon Sep 17 00:00:00 2001
-From: Even Rouault <even.rouault@spatialys.com>
-Date: Wed, 15 Feb 2023 19:56:16 +0100
-Subject: [PATCH] Only use absl headers if C++14 is available. (#7824)
-
-If flatbuffers is built in C++11 mode, but there is a recent version of
-absl which requires C++14, the build will fail.
-Cf https://github.com/MapServer/MapServer/issues/6822 for the use case
-that triggered this.
----
- include/flatbuffers/base.h | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
 diff --git a/include/flatbuffers/base.h b/include/flatbuffers/base.h
 index 86688cc6e40..219b6d308ac 100644
 --- a/include/flatbuffers/base.h

--- a/ports/flatbuffers/fix-abseil-cxx14.patch
+++ b/ports/flatbuffers/fix-abseil-cxx14.patch
@@ -1,0 +1,26 @@
+From a56f9ec50e908362e20254fcef28e62a2f148d91 Mon Sep 17 00:00:00 2001
+From: Even Rouault <even.rouault@spatialys.com>
+Date: Wed, 15 Feb 2023 19:56:16 +0100
+Subject: [PATCH] Only use absl headers if C++14 is available. (#7824)
+
+If flatbuffers is built in C++11 mode, but there is a recent version of
+absl which requires C++14, the build will fail.
+Cf https://github.com/MapServer/MapServer/issues/6822 for the use case
+that triggered this.
+---
+ include/flatbuffers/base.h | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/include/flatbuffers/base.h b/include/flatbuffers/base.h
+index 86688cc6e40..219b6d308ac 100644
+--- a/include/flatbuffers/base.h
++++ b/include/flatbuffers/base.h
+@@ -233,7 +233,7 @@ namespace flatbuffers {
+       }
+       #define FLATBUFFERS_HAS_STRING_VIEW 1
+     // Check for absl::string_view
+-    #elif __has_include("absl/strings/string_view.h")
++    #elif __has_include("absl/strings/string_view.h") && (__cplusplus >= 201411)
+       #include "absl/strings/string_view.h"
+       namespace flatbuffers {
+         typedef absl::string_view string_view;

--- a/ports/flatbuffers/portfile.cmake
+++ b/ports/flatbuffers/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-uwp-build.patch
+        fix-abseil-cxx14.patch
 )
 
 set(options "")

--- a/ports/flatbuffers/vcpkg.json
+++ b/ports/flatbuffers/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "flatbuffers",
   "version": "23.1.21",
-  "port-version": 1,
+  "port-version": 2,
   "description": [
     "Memory Efficient Serialization Library",
     "FlatBuffers is an efficient cross platform serialization library for games and other memory constrained apps. It allows you to directly access serialized data without unpacking/parsing it first, while still having great forwards/backwards compatibility."

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2454,7 +2454,7 @@
     },
     "flatbuffers": {
       "baseline": "23.1.21",
-      "port-version": 1
+      "port-version": 2
     },
     "flecs": {
       "baseline": "3.1.4",

--- a/versions/f-/flatbuffers.json
+++ b/versions/f-/flatbuffers.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "fb8cde16588624e98a6f402cecbe3f178df74d3e",
+      "version": "23.1.21",
+      "port-version": 2
+    },
+    {
       "git-tree": "c8dd8a45a079d9ec27da5352d1d61eb24ff94f5d",
       "version": "23.1.21",
       "port-version": 1

--- a/versions/f-/flatbuffers.json
+++ b/versions/f-/flatbuffers.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "fb8cde16588624e98a6f402cecbe3f178df74d3e",
+      "git-tree": "8079babd92cd3c1a80b224beb8f034b1a912a8b6",
       "version": "23.1.21",
       "port-version": 2
     },

